pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -137,10 +137,29 @@ func (c *NPLController) Run(stopCh <-chan struct{}) {
 	for i := 0; i < numWorkers; i++ {
 		go wait.Until(c.Worker, time.Second, stopCh)
 	}
-
+	go c.SyncRules(stopCh)
 	<-stopCh
 }
-
+func NewTestingNPLController(start, end int)  (*NPLController, error) {
```

Comment:
would be better if this took as parameter the port table instead of `start` & `end`, so it aligns more closely with the "normal" `NewNPLController` function. This definitely needs a comment as well, since this can only be used for a certain kind of testing, and the `Run` method should not be invoked on the returned `NPLController` instance.

Reply:
Fixed